src/cryptography/hazmat/backends/openssl/x509.py

```diff
@@ -151,3 +156,66 @@ def signature_hash_algorithm(self):
             raise UnsupportedAlgorithm(
                 "Signature algorithm OID:{0} not recognized".format(oid)
             )
+
+    @property
+    def extensions(self):
+        # TODO: remove this when we finish landing support for extensions.
+        warnings.warn(
+            "Extension support is not fully implemented. Unsupported "
+            "extensions are IGNORED at this time."
+        )
```

Comment:
Alternate suggestion then, what if we emit a warning on specific critical
extensions we don't handle, and then move to eliminate those warnings ASAP
and convert to an exception?

On Fri, Mar 27, 2015 at 10:43 PM, Paul Kehrer <notifications@github.com>
wrote:

> In src/cryptography/hazmat/backends/openssl/x509.py
> <https://github.com/pyca/cryptography/pull/1769#discussion_r27341710>:
>
> > @@ -151,3 +156,66 @@ def signature_hash_algorithm(self):
> >              raise UnsupportedAlgorithm(
> >                  "Signature algorithm OID:{0} not recognized".format(oid)
> >              )
> > +
> > +    @property
> > +    def extensions(self):
> > +        # TODO: remove this when we finish landing support for extensions.
> > +        warnings.warn(
> > +            "Extension support is not fully implemented. Unsupported "
> > +            "extensions are IGNORED at this time."
> > +        )
>
> If you have the review cycles we could land KeyUsage next week no problem.
> It was my plan to do KeyUsage and ExtendedKeyUsage while SAN went through
> review.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/1769/files#r27341710>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084